src/android/ManagerService.java

```diff
@@ -0,0 +1,504 @@
+                stopService(intent);
+
+                requestObserver.unregister();
+                requestObserver = null;
+            } else {
+                Intent intent = new Intent(ManagerService.this, ManagerService.class);
+                stopService(intent);
+
+                foregroundNotification(this.inputTitle, this.inputContent);
+            }
+        }
+    }
+
+    @Nullable
+    @Override
+    public IBinder onBind(Intent intent) {
+        return mBinder;
+    }
+
+    @Override
+    public int onStartCommand(Intent intent, int flags, int startId) {
+        initUploadService();
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            try {
+                JSONObject settings = new JSONObject(intent.getStringExtra("options"));
+                this.inputTitle = settings.getString("foregroundTitle");
+                this.inputContent = settings.getString("foregroundContent");
+            } catch (JSONException error) {
+                error.getLocalizedMessage();
+            }
+
+            foregroundNotification(inputTitle, inputContent);
+        }
+
+        this.requestObserver = new GlobalRequestObserver(this.getApplication(), broadcastReceiver);
+        this.requestObserver.register();
+
+        int parallelUploadsLimit = 1;
+
+        try {
+            JSONObject settings = new JSONObject(intent.getStringExtra("options"));
+            parallelUploadsLimit = settings.getInt("parallelUploadsLimit");
+        } catch (JSONException error) {
+            ManagerService.logMessage(String.format("eventLabel='Uploader could not read parallelUploadsLimit from config' error='%s'", error.getMessage()));
+        }
+
+        UploadServiceConfig.setNotificationHandlerFactory((uploadService) -> new NotificationHandler(uploadService, mainActivity));
+
+        UploadServiceConfig.setHttpStack(new OkHttpStack());
+        ExecutorService threadPoolExecutor =
+                new ThreadPoolExecutor(
+                        parallelUploadsLimit,
+                        parallelUploadsLimit,
+                        5000,
+                        TimeUnit.MILLISECONDS,
+                        new LinkedBlockingQueue<Runnable>()
+                );
+        UploadServiceConfig.setThreadPool((AbstractExecutorService) threadPoolExecutor);
+
+        networkObservable = ReactiveNetwork
+                .observeNetworkConnectivity(this)
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(connectivity -> {
+                    logMessage(String.format("eventLabel='Uploader Network connectivity changed' connectivity_state='%s'", connectivity.state()));
+                    isNetworkAvailable = connectivity.state() == NetworkInfo.State.CONNECTED;
+                    if (isNetworkAvailable) {
+                        uploadPendingList();
+                    }
+                });
+
+        return START_NOT_STICKY;
```

Comment:
Pourquoi tu as choisis NOT_STICKY ?

Faut pas mettre STICKY et c'est le service qui s’arrête lui même ?

Quand il n'y pas de connections, est-ce que le service s’arrête ? Et comment il resume ?

Reply:
Est-ce qu'on tentera de recréé le service si il est détruit? Si c'est le cas, je pourrai le mettre en sticky

Reply:
On essaierait de re-créer le service quand SharinPix est de nouveau ouvert..
Sinon, tant qu'il y a des uploads en pending, le service doit rester actif. Quand il a terminé, il s'arrête.
Si le système l'a tué, quand SharinPix rouvre, il va le recréer.

Reply:
Il y a deux cas dans ce qui est fait jsq'ici:
- si l'application n'est pas en ouvert et qu'il n'y a pas d'uploads en cours donc le service est créé à l'ouverture de cette dernière
- si des uploads sont en cours, ceci dit le service est déjà créé; l'application s'ouvre et utilise ce service pour d'autres usages

Reply:
le case de sticky/not
- si start_sticky: il envisage de recréé le service si il est détruit due à une insuffisance de mémoire par exemple (mais pourrai causer une anr car il va envoyer une intent vide dans notre cas)
- si start_not_sticky: il ne recrée pas le service

Reply:
> Because it is in the started state, it will guarantee to call onStartCommand(Intent, int, int) after creating the new service instance; if there are not any pending start commands to be delivered to the service, it will be called with a null intent object, so you must take care to check for this. https://developer.android.com/reference/android/app/Service#START_STICKY

@sc-nick si c'est start_sticky et le système relance le service, les uploads vont continuer, n'est-ce pas ? On aura déjà les infos nécessaires pour continuer j'espère... Et donc on ignore l'intent vide et tout continue ?

À la limite pour commencer, on pourrait mettre NOT_STICKY - ça ne me derange pas. Si on a des problèmes on regarde pour STICKY. @ombr ?

Reply:
theoriquement oui